Foxtrot/Driver/Rewriting/PostRewriteChecker.cs

```diff
@@ -0,0 +1,169 @@
+        private readonly Stack<Member> visitedMembers;
+
+        /// <summary>
+        ///     Holds a read-only reference to an error handler.
+        /// </summary>
+        private readonly Action<CompilerError> handleError;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="PostRewriteChecker"/> class with
+        ///     a reference to an action responsible for handling reported errors.
+        /// </summary>
+        /// <param name="handleError">
+        ///     An action that will be called for each error occurred.
+        /// </param>
+        public PostRewriteChecker(Action<CompilerError> handleError)
+        {
+            Contract.Requires(handleError != null);
+            Contract.Ensures(object.ReferenceEquals(this.handleError, handleError));
+
+            this.handleError = handleError;
+            this.visitedMembers = new Stack<Member>();
+        }
+
+        /// <summary>
+        ///     A method that gets called for each property definition in analyzed assembly.
+        /// </summary>
+        /// <param name="member">
+        ///     An object that represents a property.
+        /// </param>
+        public override void VisitProperty(Property member)
+        {
+            this.visitedMembers.Push(member);
+
+            base.VisitProperty(member);
+
+            this.visitedMembers.Pop();
+        }
+
+        /// <summary>
+        ///     A method that gets called for each event definition in analyzed assembly.
+        /// </summary>
+        /// <param name="member">
+        ///     An object that represents an event.
+        /// </param>
+        public override void VisitEvent(Event member)
+        {
+            this.visitedMembers.Push(member);
+
+            base.VisitEvent(member);
+
+            this.visitedMembers.Pop();
+        }
+
+        /// <summary>
+        ///     A method that gets called for each method definition in analyzed assembly.
+        /// </summary>
+        /// <param name="member">
+        ///     An object that represents a method.
+        /// </param>
+        public override void VisitMethod(Method member)
+        {
+            this.visitedMembers.Push(member);
+
+            base.VisitMethod(member);
+
+            this.visitedMembers.Pop();
+        }
+
+        /// <summary>
+        ///     A method that gets called for each type's member binding used in analyzed assembly.
+        /// </summary>
+        /// <param name="binding">
+        ///     An object that represents a member binding.
+        /// </param>
+        /// <remarks>
+        ///     This method reports error for each call to Code Contracts' method which normally should be rewritten.
+        /// </remarks>
+        public override void VisitMemberBinding(MemberBinding binding)
+        {
+            base.VisitMemberBinding(binding);
+
+            if (object.ReferenceEquals(binding, null) ||
+                this.visitedMembers.Count == 0 ||
```

Comment:
So it means that this check ensures that only for those type we need to analyze current bound member represented by `binding`?